packages/hebao_v1/contracts/modules/dapps/CompoundModule.sol

```diff
@@ -22,24 +22,32 @@ import "../../base/BaseSubAccount.sol";
 import "../../thirdparty/compound/CompoundRegistery.sol";
 import "../../thirdparty/compound/CEther.sol";
 import "../../thirdparty/compound/CErc20.sol";
+import "../../thirdparty/compound/Comptroller.sol";
 
 import "../security/SecurityModule.sol";
 
 
 /// @title CompoundModule
 contract CompoundModule is BaseSubAccount, SecurityModule
 {
+    event CollateralAdded(address indexed _wallet, address _collateral, uint _collateralAmount);
+    event CollateralRemoved(address indexed _wallet, address _collateral, uint _collateralAmount);
+    event SubAccountBorrow(address indexed _wallet, address _debtToken, int _debtAmount);
+
     CompoundRegistry internal compoundRegistry;
+    Comptroller      internal comptroller;
     address constant internal ETH_TOKEN_ADDRESS = address(0);
 
     constructor(
         Controller       _controller,
-        CompoundRegistry _compoundRegistry
+        CompoundRegistry _compoundRegistry,
+        Comptroller      _comptroller
         )
         public
         SecurityModule(_controller)
     {
         compoundRegistry = _compoundRegistry;
+        comptroller = _comptroller;
     }
 
     /// @dev Fund Compound for earn interests and automatically enters market
```

Comment:
I think this comment is wrong as the market is only exited when there is no loan for the token and balanceOf(cToken) != 0. The market is never entered.

Reply:
Fixed